library/packages/src/lib/y2packager/product_reader.rb

```diff
@@ -78,30 +79,45 @@ def all_products
         []
       end
 
-      @all_products ||= available_products.each_with_object([]) do |prod, all_products|
-        prod_pkg = product_package(prod["product_package"])
+      return @all_products if @all_products
 
-        if prod_pkg
-          # remove special products if they have not been defined in linuxrc
-          prod_pkg["deps"].find { |dep| dep["provides"] =~ /\Aspecialproduct\(\s*(.*?)\s*\)\z/ }
-          special_product_tag = linuxrc_string(Regexp.last_match[1]) if Regexp.last_match
-          if special_product_tag && !linuxrc_special_products.include?(special_product_tag)
-            log.info "Special product #{prod["name"]} has not been defined via linuxrc. --> do not offer it"
-            next
-          end
+      @all_products = []
 
-          # Evaluating display order
-          prod_pkg["deps"].find { |dep| dep["provides"] =~ /\Adisplayorder\(\s*([0-9]+)\s*\)\z/ }
-          displayorder = Regexp.last_match[1].to_i if Regexp.last_match
+      if Yast::Stage.initial && Y2Packager::MediumType.online?
+        Y2Packager::ProductControlProduct.products.each do |p|
+          @all_products << Y2Packager::Product.new(name: p.name, display_name: p.label,
+            version: p.version,
+            installation_package: "Test" # just hack as we do not know current package name yet
```

Comment:
sadly, this does not work, as it is needed later when trying to merge workflows and package is needed.
In general problem is that until we migrate, we do not have base products and after that step we need to load it from real repo from scc, but at many places there is caching.

Reply:
NP: maybe something like "Not known yet" could be better here. But it does not matter much...